docs/Design/Exceptions/Exception Handling.md

```diff
@@ -0,0 +1,286 @@
+
+### `catch end`
+
+`catch end`
+
+Ends a catch block.
+
+### `finally start`
+
+`finally start`
+
+Starts a finally block.
+
+### `finally end`
+
+`finally end`
+
+Ends a finally block.
+
+## Usage
+
+```
+try start
+    comment : Do stuff
+try end
+catch start : exceptionType foo
+    comment : Sometimes do stuff
+catch end
+finally start
+    comment : Always do stuff
+finally end
+```
+
+### CSharp
+
+```csharp
+try
+{
+    // Do stuff
+}
+catch (exceptionType foo)
+{
+    // Sometimes do stuff
+}
+finally
+{
+    // Always do stuff
+}
+```
+
+### Java
+
+```java
+try {
+    // Do stuff
+} catch (exceptionType foo) {
+    // Sometimes do stuff
+} finally {
+    // Always do stuff
+}
+```
+
+### Python
+
+```python
+try:
+    ## Do stuff
+except exceptionType as foo:
+    ## Sometimes do stuff
+finally:
+    ## Always do stuff
+```
+
+### Ruby
+
+```ruby
+begin
+    ## Do stuff
+rescue exceptionType => foo
+    ## Sometimes do stuff
+ensure
+    ## Always do stuff
+```
+
+### TypeScript
+
+```typescript
+try {
+    // Do stuff
+} catch (exceptionType foo) {
+    // Sometimes do stuff
+} finally {
+    // Always do stuff
+}
+```
+
+## Implementation
+
+### Properties
+
+<table>
+    <thead>
+        <th>Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>TryStartLeft</td>
+            <td><code>string</code></td>
+            <td>The beginning of try start expression</td>
+        </tr>
+        <tr>
+            <td>TryStartRight</td>
+            <td><code>string</code></td>
+            <td>The end of try start expression</td>
+        </tr>
+        <tr>
+            <td>CatchStartLeft</td>
+            <td><code>string</code></td>
+            <td>The beginning of catch start expression</td>
+        </tr>
+        <tr>
+            <td>CatchStartMiddle</td>
+            <td><code>string</code></td>
+            <td>The middle of catch start expression</td>
+        </tr>
+        <tr>
+            <td>CatchStartLink</td>
+            <td><code>string</code></td>
+            <td>Links the alias and the exceptionType arguments</td>
+        </tr>
+        <tr>
+            <td>CatchStartRight</td>
+            <td><code>string</code></td>
+            <td>The end of catch start expression</td>
+        </tr>
+        <tr>
+            <td>FinallyStartLeft</td>
+            <td><code>string</code></td>
+            <td>The beginning of finally start expression</td>
+        </tr>
+        <tr>
+            <td>FinallyStartRight</td>
+            <td><code>string</code></td>
+            <td>The end of finally start expression</td>
+        </tr>
+        <tr>
+            <td>BlockEnd</td>
+            <td><code>string</code></td>
+            <td>The end of try, catch, and finally blocks</td>
+        </tr>
+    </tbody>
+</table>
+
+### Langauge Values
+
+#### Table 1
```

Comment:
Instead of arbitrarily splitting the property values into "Table 1" and "Table 2", split them on something relevant, like try/catch/finally tables.